### DISCUSSION TITLE
Suppress only some types of warnings, e.g. for `vspace`

### DISCUSSION THREAD

**Discussion by c-bg** [James-Yu/LaTeX-Workshop#3381] [Q&A] [answered]
**tl: dr**: I'd like it if one could suppress a specific type of warning, in particular warnings for using `\vspace`. 
I checked the wiki but only found an [option for disabling all warning boxes](https://github.com/James-Yu/LaTeX-Workshop/wiki/Compile#latex-workshopmessagewarningshow).

## To Reproduce 

1. use the `\vspace{}` command 
2. see warning/problem box with `\vspace should only be used to provide space above/below surrounding objects.`

## Describe the solution you'd like

An option for suppressing this type of warning for an entire project, or for a specific line. 
possible implementation for the latter:
```latex
%  latex-workshop-suppress: vspace   % suppresses the warning for the following line. <vspace> can be replaced with the concrete warning name
\vspace{-1cm} asdf
```

Thanks a lot for maintaining this extension, I find it very useful!  :)

**Comment by jlelong** [comment, marked as answer]
These warnings are generated by the linter `chktex`. To suppress specific warnings, please refer to `chktex` documentation.

**Reply by c-bg** [reply]
Thanks so much for your prompt and helpful reply. 
Perfect, I see, I'll look into this.

**Reply by c-bg** [reply]
For anyone wondering about the same, here's what you can do. 

1. **Identify the warning number that should be suppressed**   
   You can find a list of all warning numbers/warnings in the `chktex` [docs](https://mirror.foobar.to/CTAN/systems/doc/chktex/ChkTeX.pdf). 

2. **Open `settings.json`**  
    In VS Code, hit `ctrl+shift+p`/mac: `cmd+shift+p` -> `Preferences: Open Settings (JSON)` 

3. **Call `chktex` with warning suppression** 
   Tell  LaTeX-Workshop to call `chktex` with the `-n <warning-number>` argument (`-n` = `--no-warn`), additionally to the [default arguments](https://github.com/James-Yu/LaTeX-Workshop/wiki/Linters#latex-workshopchktexexecargsactive) `["-wall", "-n22", "-n30", "-e16", "-q"]`. You can do this by adding the following line inside the outermost pair of `{  }` brackets in the JSON; 
 simply replace the warning number with the one you found before.
   ```json
   "latex-workshop.chktex.exec.args": ["-wall", "-n22", "-n30", "-e16", "-q", "-n", "<warning-number>"],
   ```
